src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1257,12 +1257,15 @@ CopyLocalDataIntoShards(Oid distributedRelationId)
 	ExprContext *econtext = GetPerTupleExprContext(estate);
 	econtext->ecxt_scantuple = slot;
 
+	/* here we already have the data locally */
```

Comment:
If we have the data locally, I thought we can use the local copy, why we are setting it to `false`?

Reply:
At this point we already have the data in the local node and we are going to copy the data to workers. So without this, we will copy the data locally even though we already have it. 

I think the naming is confusing here, maybe I can rename it to `hasCopyDataLocally`.

Reply:
Why do we need this flag? Does it mean create_distributed_table always makes connections to localhost? That seems off.

Reply:
ping

Reply:
So it seems that the issue was that we were creating the shards over a connection for localhost as well and it wouldnt be visible when we were trying the copy the data locally to those shards.

The problem was that we should not do local copy if `TransactionConnectedToLocalGroup`  is true.